### PR TITLE
[spinel] make `kTxWaitUs` configurable during compile time

### PR DESCRIPTION
**Pull request by parag-silabs**
It is useful if RCP needs to wait longer than the hardcoded value in kTxWaitUs.

### CODE REVIEW

File: src/lib/spinel/radio_spinel.hpp [hunk 1]
```diff
@@ -1107,7 +1107,7 @@ class RadioSpinel
 
     static constexpr uint32_t kUsPerMs = 1000; ///< Microseconds per millisecond.
     static constexpr uint64_t kTxWaitUs =
-        5000000; ///< Maximum time of waiting for `TransmitDone` event, in microseconds.
+        OPENTHREAD_SPINEL_CONFIG_RCP_TX_WAIT_TIME_SECS * kUsPerMs * 1000; ///< Maximum time of waiting for `TransmitDone` event, in microseconds.
```
Comment (abtink) [inline src/lib/spinel/radio_spinel.hpp:1110]: We use `kUsPerMs` and then hard-coded `* 1000` for the sec to msec conversion part?
Should we define a `kUsPerSec` for this?

Comment (parag-silabs) [reply src/lib/spinel/radio_spinel.hpp]: Thanks Abtin, for reviewing the changes.

Make sense to define `kUsPerSec`, please have a look at the updated changes.

**Review by abtink — APPROVED** [review]
LGTM. 👍

**Comment by parag-silabs** [comment]
@jwhui Can you please review the PR?

File: etc/cmake/options.cmake [hunk 2]
```diff
@@ -325,6 +325,8 @@ ot_int_option(OT_RCP_RESTORATION_MAX_COUNT OPENTHREAD_SPINEL_CONFIG_RCP_RESTORAT
 
 # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 
+ot_int_option(OT_RCP_TX_WAIT_TIME_SECS  OPENTHREAD_SPINEL_CONFIG_RCP_TX_WAIT_TIME_SECS "set RCP TX wait TIME in seconds")
```
Comment (jwhui) [inline etc/cmake/options.cmake:328]: nit:
```suggestion
ot_int_option(OT_RCP_TX_WAIT_TIME_SECS OPENTHREAD_SPINEL_CONFIG_RCP_TX_WAIT_TIME_SECS "set RCP TX wait TIME in seconds")
```

File: script/cmake-build [hunk 3]
```diff
@@ -107,6 +107,7 @@ OT_POSIX_SIM_COMMON_OPTIONS=(
     "-DOT_SNTP_CLIENT=ON"
     "-DOT_SRP_CLIENT=ON"
     "-DOT_SRP_SERVER=ON"
+    "-DOT_RCP_TX_WAIT_TIME_SECS=5"
```
Comment (jwhui) [inline script/cmake-build:110]: nit: maintain alphabetical order

Comment (parag-silabs) [reply script/cmake-build]: I did not realize earlier that they were alphabetized. Thanks for catching it.

File: etc/cmake/options.cmake [hunk 4]
```diff
@@ -325,6 +325,8 @@ ot_int_option(OT_RCP_RESTORATION_MAX_COUNT OPENTHREAD_SPINEL_CONFIG_RCP_RESTORAT
 
 # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 
+ot_int_option(OT_RCP_TX_WAIT_TIME_SECS OPENTHREAD_SPINEL_CONFIG_RCP_TX_WAIT_TIME_SECS "set RCP TX wait TIME in seconds")
```
Comment (jwhui) [inline etc/cmake/options.cmake:328]: Another nit: colocate this with `OT_RCP_RESTORATION_MAX_COUNT` above?

**Review by jwhui — APPROVED** [review]

**Pull request merged**